### PR TITLE
tools/libressl: update to 3.8.2

### PR DESCRIPTION
**Pull request by misaka36**
Release Notes:
https://ftp.openbsd.org/pub/OpenBSD/LibreSSL/libressl-3.8.0-relnotes.txt

### CODE REVIEW

**Comment by lovehackintosh** [comment]
```
====== Make errors from logs/tools/libressl/compile.txt ======
logs/tools/libressl/compile.txt-  CC       gost/libcrypto_la-gost89_keywrap.lo
logs/tools/libressl/compile.txt-In file included from gost/gost2814789.c:437:
logs/tools/libressl/compile.txt-In file included from ./md32_common.h:114:
logs/tools/libressl/compile.txt-./crypto_internal.h:34:6: error: implicit declaration of function 'htobe32' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
logs/tools/libressl/compile.txt-        v = htobe32(v);
logs/tools/libressl/compile.txt-            ^
logs/tools/libressl/compile.txt-./crypto_internal.h:48:6: error: implicit declaration of function 'htobe64' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
logs/tools/libressl/compile.txt-        v = htobe64(v);
logs/tools/libressl/compile.txt-            ^
logs/tools/libressl/compile.txt-2 errors generated.
logs/tools/libressl/compile.txt:make[5]: *** [Makefile:8017: gost/libcrypto_la-gost2814789.lo] Error 1
logs/tools/libressl/compile.txt-make[5]: *** Waiting for unfinished jobs....
logs/tools/libressl/compile.txt-make[5]: Leaving directory '/Volumes/OpenWrt/openwrt/build_dir/host/libressl-3.8.0/crypto'
logs/tools/libressl/compile.txt:make[4]: *** [Makefile:2259: all] Error 2
logs/tools/libressl/compile.txt-make[4]: Leaving directory '/Volumes/OpenWrt/openwrt/build_dir/host/libressl-3.8.0/crypto'
logs/tools/libressl/compile.txt:make[3]: *** [Makefile:4[60](https://github.com/openwrt/openwrt/actions/runs/5161809533/jobs/9299085512?pr=12799#step:6:61): all-recursive] Error 1
logs/tools/libressl/compile.txt-make[3]: Leaving directory '/Volumes/OpenWrt/openwrt/build_dir/host/libressl-3.8.0'
logs/tools/libressl/compile.txt:make[2]: *** [Makefile:41: /Volumes/OpenWrt/openwrt/build_dir/host/libressl-3.8.0/.built] Error 2
logs/tools/libressl/compile.txt-time: tools/libressl/compile#76.45#48.34#269.85
```

File: tools/include/endian.h [hunk 1]
```diff
@@ -10,6 +10,14 @@
 #define bswap_16(x) OSSwapInt16(x)
 #define bswap_32(x) OSSwapInt32(x)
 #define bswap_64(x) OSSwapInt64(x)
+#define be16toh(x) OSSwapBigToHostInt16((x))
+#define htobe16(x) OSSwapHostToBigInt16((x))
+#define be32toh(x) OSSwapBigToHostInt32((x))
```
Comment (PolynomialDivision) [inline tools/include/endian.h:15]: Why do you have here `(( ))`? So double brackets?

Comment (PolynomialDivision) [reply tools/include/endian.h:15]: ping @lovehackintosh

Comment (lovehackintosh) [reply tools/include/endian.h:15]: I just copied it from https://github.com/libressl/portable/blob/master/include/compat/endian.h#L54

Comment (lovehackintosh) [reply tools/include/endian.h]: Removed double brackets, thank you.

File: tools/include/endian.h [hunk 2]
```diff
@@ -10,6 +10,14 @@
...
 #define bswap_32(x) OSSwapInt32(x)
 #define bswap_64(x) OSSwapInt64(x)
+#define be16toh(x) OSSwapBigToHostInt16(x)
+#define htobe16(x) OSSwapHostToBigInt16(x)
+#define be32toh(x) OSSwapBigToHostInt32(x)
+#define htobe32(x) OSSwapHostToBigInt32(x)
+#define htole64(x) OSSwapHostToLittleInt64(x)
+#define htobe64(x) OSSwapHostToBigInt64(x)
+#define le64toh(x) OSSwapLittleToHostInt64(x)
+#define be64toh(x) OSSwapBigToHostInt64(x)
```
Comment (hauke) [inline tools/include/endian.h:20]: @nbd168  has similar changes in his tree, see: https://git.openwrt.org/?p=openwrt/staging/nbd.git;a=commitdiff;h=33e84e8eef15d0eb13520669963af7972cdf81cb
https://git.openwrt.org/?p=openwrt/staging/nbd.git;a=shortlog

@nbd168  could you please have a look at this.

Comment (lovehackintosh) [reply tools/include/endian.h]: Looks fine, applied.

**Comment by hauke** [comment]
@nbd168  merged some changes in 169757c7fd483787fb910c9d7b366fcd91ead49f, please rebase and let the CI run again.

**Comment by PolynomialDivision** [comment]
I just rebased on main. @lovehackintosh Do you have time to look into this again? Would be nice to have up-to-date versions of a crypto stack (at least from my view).

**Comment by misaka36** [comment]
I made a patch to fix mkimage build error. @PolynomialDivision Do you have time to test this?

**Comment by PolynomialDivision** [comment]
> I made a patch to fix mkimage build error. @PolynomialDivision Do you have time to test this?

Did you send the patch upstream?

**Comment by misaka36** [comment]
No, I need more tests before sending the patch.

**Comment by PolynomialDivision** [comment]
Before merging that here, it would be nice to get feedback from the upstream project. That is why I asked.

**Comment by sppmasterspp** [comment]
For me the compilation of mkimage still fails with 3.9.2.

**Comment by Neustradamus** [comment]
LibreSSL 3.9.2 PR is here:
- https://github.com/openwrt/openwrt/pull/15582

It is possible to look?

**Comment by Neustradamus** [comment]
@misaka36: A PR for LibreSSL 4.0.0 is here:
- https://github.com/openwrt/openwrt/pull/16901

**Pull request closed without merging**